nautobot_chatops/workers/nautobot.py

```diff
@@ -1045,6 +1051,343 @@ def get_circuit_providers(dispatcher, *args):
+    except Job.DoesNotExist:
+        dispatcher.send_error(f"Job {job_name} not found")
+        return (CommandStatusChoices.STATUS_FAILED, f'Job "{job_name}" was not found')
+
+    if not job_model.enabled:
+        dispatcher.send_error(f"The requested job {job_name} is not enabled")
+        return (CommandStatusChoices.STATUS_FAILED, f'Job "{job_name}" is not enabled')
+
+    form_class = get_job(job_model.class_path).as_form()
+
+    # Parse base form fields from job class
+    form_fields = []
+    for field_name, _ in form_class.base_fields.items():  # pylint: disable=unused-variable
+        if field_name.startswith("_"):
+            continue
+        form_fields.append(f"{field_name}")
+
+    # Basic logic check with what we know, we should expect run-job-form vs run-job to parse the same base fields
+    if len(form_fields) != len(args):
+        dispatcher.send_error("The form class fields and the passed run-job args do not match.")
+        return (
+            CommandStatusChoices.STATUS_FAILED,
+            "The form class fields and the passed run-job args do not match.",
+        )
+
+    form_item_kwargs = {}
+    for index, _ in enumerate(form_fields):  # pylint: disable=unused-variable
+        # Check request input (string-type) is also valid JSON
+        if args[index][0] == "{":
+            try:
+                json_arg = json.loads(args[index])
+                if not json_arg.get("id"):
+                    dispatcher.send_error("Form field arg is JSON dictionary, and has no `id` key.")
+                    return (
+                        CommandStatusChoices.STATUS_FAILED,
+                        "Form field arg is JSON dictionary, and has no `id` key.",
+                    )
+                form_item_kwargs[form_fields[index]] = json_arg.get("id")
+                continue
+            except json.JSONDecodeError:
+                form_item_kwargs[form_fields[index]] = args[index]
+                continue
+        form_item_kwargs[form_fields[index]] = args[index]
+
+    job_result = JobResult.enqueue_job(
+        job_model=job_model,
+        user=dispatcher.user,
+        profile=profile,
+        **form_item_kwargs,
+    )
+
+    # Wait on the job to finish
+    max_wait_iterations = 60
+    while job_result.status not in JobResultStatusChoices.READY_STATES:
+        max_wait_iterations -= 1
+        if not max_wait_iterations:
+            dispatcher.send_error(f"The requested job {job_name} failed to reach ready state.")
+            return (CommandStatusChoices.STATUS_FAILED, f'Job "{job_name}" failed to reach ready state.')
+        time.sleep(1)
+        job_result.refresh_from_db()
+
+    if job_result and job_result.status == "FAILURE":
+        dispatcher.send_error(f"The requested job {job_name} failed to initiate. Result: {job_result.result}")
+        return (CommandStatusChoices.STATUS_FAILED, f'Job "{job_name}" failed to initiate. Result: {job_result.result}')
```

Comment:
This is checking if the task failed, but then sending a message that the task failed to initiate. The task was initiated but failed.

Reply:
Good point, I adjusted the phrasing in [875ca64](https://github.com/nautobot/nautobot-app-chatops/pull/270/commits/875ca649bdd0cde11e6ac7a0d668ee729560f463)